src/lem/mod.rs

```diff
@@ -192,10 +185,13 @@ impl LEM {
     /// Instantiates a `LEM` with the appropriate transformations to make sure
     /// that constraining will be smooth.
     pub fn new(input: [Var; 3], lem: &LEMCTL) -> Result<LEM> {
-        let mut map = HashMap::from_iter(input.iter().map(|i| (i.clone(), i.clone())));
+        let mut map = input.iter().fold(VarMap::new(), |mut acc, i| {
+            acc.insert(i.clone(), i.clone());
```

Comment:
You could use `into_iter` and avoid one clone.

Reply:
Will change

Reply:
oh, yes, I missed the return of iter later, my bad.